src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs

```diff
@@ -0,0 +1,347 @@
+
+            var value = tm.Marshal(5);
+
+            Assert.Equal(5.0, ((NumberValue)value).Value);
+        }
+
+        [Fact]
+        public void Nesting()
+        {
+            // Be sure to use 'var' instead of 'object' so that we have compiler-time access to fields.           
+            var oneObj = new
+            {
+                data1 = "one",
+                two = new
+                {
+                    data2 = "two",
+                    three = new
+                    {
+                        data3 = "three"
+                    }
+                }
+            };
+
+            var cache = new TypeMarshallerCache();
+            var t = cache.GetMarshaler(oneObj.GetType());
+
+            var x = t.Marshal(oneObj);
+
+            var engine = new RecalcEngine();
+            engine.UpdateVariable("x", x);
+
+            var result1 = engine.Eval("x.two.three.data3");
+            Assert.Equal("three", ((StringValue)result1).Value);
+        }
+
+        private class TestNode
+        {
+            public int Data { get; set; }
+
+            public TestNode Next { get; set; }
+        }
+
+        [Fact]
+        public void TestBlank()
+        {
+            var node1 = new TestNode
+            {
+                Data = 10,
+                Next = null
+            };
+
+            var cache = new TypeMarshallerCache();
+            var x = cache.Marshal(node1);
+
+            var engine = new RecalcEngine();
+            engine.UpdateVariable("x", x);
+
+            var result1 = engine.Eval("x.Next");
+            Assert.IsType<BlankValue>(result1);
+        }
+
+        [Fact]
+        public void TestRecursion()
+        {
+            // Be sure to use 'var' instead of 'object' so that we have compiler-time access to fields.           
+            var node1 = new TestNode
+            {
+                Data = 10,
+                Next = new TestNode
+                {
+                    Data = 20,
+                    Next = new TestNode
+                    {
+                        Data = 30
+                    }
+                }
+            };
+
+            // create a cycle. 
+            // Recursion has a default marshalling depth. 
+            node1.Next.Next.Next = node1;
+
+            var cache = new TypeMarshallerCache();
+            var x = cache.Marshal(node1);
+
+            // If we made it here, at leat we didn't hang marshalling the infinite cycle. 
+
+            var engine = new RecalcEngine();
+            engine.UpdateVariable("x", x);
+
+            var result1 = engine.Eval("x.Data");
+            Assert.Equal(10.0, ((NumberValue)result1).Value);
+
+            var result2 = engine.Eval("x.Next.Data");
+            Assert.Equal(20.0, ((NumberValue)result2).Value);
+
+            var result3 = engine.Eval("x.Next.Next.Data");
+            Assert.Equal(30.0, ((NumberValue)result3).Value);
+
+            // $$$ test that it was Truncated...
```

Comment:
>// $$$ test that it was Truncated... [](http://example.com/codeflow?start=12&length=36)

What should be the behavior here? Set any object reference which we previously saw to null?